src/main/scala/org/hammerlab/guacamole/variants/Genotype.scala

```diff
@@ -1,28 +1,19 @@
 package org.hammerlab.guacamole.variants
 
-import com.esotericsoftware.kryo.io.{Input, Output}
-import com.esotericsoftware.kryo.{Kryo, Serializer}
-import org.bdgenomics.formats.avro.GenotypeAllele
-
 /**
- * A Genotype is a sequence of alleles of length equal to the ploidy of the organism.
+ * A Genotype is a map of alleles to their allele frequency
  *
  * A Genotype is for a particular reference locus. Each allele gives the base(s) present on a chromosome at that
  * locus.
  *
- * For example, the possible single-base diploid genotypes are Seq('A', 'A'), Seq('A', 'T') ... Seq('T', 'T').
- * Alleles can also be multiple bases as well, e.g. Seq("AAA", "T")
- *
  */
-case class Genotype(alleles: Allele*) {
-  /**
-   * The ploidy of the organism is the number of alleles in the genotype.
-   */
-  val ploidy = alleles.size
+case class Genotype(alleleMixture: Map[Allele, Double]) {
 
-  lazy val uniqueAllelesCount = alleles.toSet.size
+  assume(alleleMixture.values.sum <= 1, "Allele fractions are larger than 1")
```

Comment:
Yes, this test was originally `= 1` but was thinking it might be more likely that we assume `.5` for the reference and then something like `.2` for the variant if it is a low purity sample. Could possibly come back and adjust this.